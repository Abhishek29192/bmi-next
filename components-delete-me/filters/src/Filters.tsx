import React, { useContext } from "react";
import DefaultCheckbox from "@bmi-digital/components/checkbox";
import Typography from "@bmi-digital/components/typography";
import {
  getMicroCopy,
  MicroCopyContext
} from "@bmi-digital/components/micro-copy";
import Accordion from "@bmi-digital/components/accordion";
import styles from "./Filters.module.scss";

type FilterOption = {
  label: React.ReactNode;
  value: string;
  isDisabled?: boolean;
};

export type Filter = {
  label: string;
  name: string;
  value?: ReadonlyArray<string>;
  options: ReadonlyArray<FilterOption>;
};

export type Props = {
  filters: ReadonlyArray<Filter>;
  onChange?: (filterName: string, filterValue: string, value: boolean) => void;
  checkboxComponent?: React.ComponentType<any>; // TODO
  accordionSummaryComponent?: React.ComponentType<any>; // TODO
};

const Filters = ({
  filters,
  onChange,
  checkboxComponent: Checkbox = DefaultCheckbox,
  accordionSummaryComponent: AccordionSummary = Accordion.Summary
}: Props) => {
  const copy = useContext(MicroCopyContext);

  const handleCheckboxChange: Props["onChange"] = (...args) => {
    onChange && onChange(...args);
  };

  return (
    <div className={styles["Filters"]}>
      <Accordion noInnerPadding>
        {filters.map((filter) => {
          const filterOptions = filter.options.map(
            ({ value, label, isDisabled }) => {
              const isChecked = (filter.value || []).includes(value);
              return {
                value,
                label,
                checked: isChecked,
                isDisabled
              };
            }
          );

          const summaryLabel = getMicroCopy(
            copy,
            filter.label || filter.name,
            {},
            !filter.label
          );

          return (
            <Accordion.Item key={filter.name}>
              <AccordionSummary aria-label={summaryLabel}>
                <Typography variant="h6">{summaryLabel}</Typography>
              </AccordionSummary>
              <Accordion.Details noInnerPadding={true}>
                <div className={styles["list"]}>
                  {filterOptions.map((option) => (
                    <div key={option.value}>
                      <Checkbox
                        key={option.value}
                        name={option.value}
                        label={option.label}
                        checked={option.checked}
                        disabled={option.isDisabled}
                        // TODO: withFormControl overrides onChange type, this should be boolean
                        onChange={(value: boolean) => {
                          handleCheckboxChange(
                            filter.name,
                            option.value,
                            value
                          );
                        }}
                      />
                    </div>
                  ))}
                </div>
              </Accordion.Details>
            </Accordion.Item>
          );
        })}
      </Accordion>
    </div>
  );
};

export default Filters;
