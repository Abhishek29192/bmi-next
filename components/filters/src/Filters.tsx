import React, { useContext } from "react";
import DefaultCheckbox from "@bmi/checkbox";
import Typography from "@bmi/typography";
import MicroCopy from "@bmi/micro-copy";
import Accordion from "@bmi/accordion";
import styles from "./Filters.module.scss";

type FilterOption = {
  label: React.ReactNode;
  value: string;
  isDisabled?: boolean;
};

type Filter = {
  label: React.ReactNode;
  name: string;
  value?: ReadonlyArray<string>;
  options: ReadonlyArray<FilterOption>;
};

export type Props = {
  filters: ReadonlyArray<Filter>;
  microcopyProvider: Record<string, string>;
  onChange?: (filterName: string, filterValue: string, value: boolean) => void;
  checkboxComponent?: React.ComponentType<any>; // TODO
  accordionSummaryComponent?: React.ComponentType<any>; // TODO
};

const Filters = ({
  filters,
  microcopyProvider,
  onChange,
  checkboxComponent: Checkbox = DefaultCheckbox,
  accordionSummaryComponent: AccordionSummary = Accordion.Summary
}: Props) => {
  const handleCheckboxChange: Props["onChange"] = (...args) => {
    onChange && onChange(...args);
  };

  return (
    <div className={styles["Filters"]}>
      <Accordion noInnerPadding>
        {filters.map((filter) => {
          const filterOptions = filter.options.map(
            ({ value, label, isDisabled }) => {
              const isChecked = (filter.value || []).includes(value);
              return {
                value,
                label,
                checked: isChecked,
                isDisabled
              };
            }
          );

          return (
            <Accordion.Item key={filter.name}>
              <Accordion.Summary>
                <Typography variant="h6">
                  <MicroCopy.Provider values={microcopyProvider}>
                    <MicroCopy path={filter.label.toString()} />
                  </MicroCopy.Provider>
                </Typography>
              </Accordion.Summary>
              <Accordion.Details>
                <div className={styles["list"]}>
                  {filterOptions.map((option) => (
                    <div key={option.value}>
                      <Checkbox
                        key={option.value}
                        name={option.value}
                        label={option.label}
                        checked={option.checked}
                        disabled={option.isDisabled}
                        // TODO: withFormControl overrides onChange type, this should be boolean
                        onChange={(value: boolean) => {
                          handleCheckboxChange(
                            filter.name,
                            option.value,
                            value
                          );
                        }}
                      />
                    </div>
                  ))}
                </div>
              </Accordion.Details>
            </Accordion.Item>
          );
        })}
      </Accordion>
    </div>
  );
};

export default Filters;
