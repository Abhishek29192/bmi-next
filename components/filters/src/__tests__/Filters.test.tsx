import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Filters from "../";

describe("Filters component", () => {
  it("renders correctly", () => {
    const { container } = render(<Filters filters={[]} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders multiple filters", () => {
    const filters = [
      {
        label: "Label 1",
        name: "product-1",
        options: [
          {
            label: "Option 1",
            value: "option-1"
          },
          {
            label: "Option 2",
            value: "option-2"
          }
        ]
      },
      {
        label: "Label 2",
        name: "product-2",
        options: [
          {
            label: "Option 1",
            value: "option-1"
          },
          {
            label: "Option 2",
            value: "option-2"
          }
        ]
      }
    ];

    const { container } = render(<Filters filters={filters} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders selected filters", () => {
    const filters = [
      {
        label: "Label 1",
        name: "product-1",
        value: ["option-2"],
        options: [
          {
            label: "Option 1",
            value: "option-1"
          },
          {
            label: "Option 2",
            value: "option-2"
          }
        ]
      }
    ];

    const { container } = render(<Filters filters={filters} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders disabled filters", () => {
    const filters = [
      {
        label: "Label 1",
        name: "product-1",
        value: ["option-2"],
        options: [
          {
            label: "Option 1",
            value: "option-1"
          },
          {
            label: "Option 2",
            value: "option-2",
            isDisabled: true
          }
        ]
      }
    ];

    const { container } = render(<Filters filters={filters} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onChange handler", () => {
    const filters = [
      {
        label: "Label 1",
        name: "product-1",
        value: ["option-3"],
        options: [
          {
            label: "Option 1",
            value: "option-1"
          },
          {
            label: "Option 2",
            value: "option-2",
            isDisabled: true
          },
          {
            label: "Option 3",
            value: "option-3"
          }
        ]
      }
    ];
    const onChange = jest.fn();

    const { getByLabelText } = render(
      <Filters filters={filters} onChange={onChange} />
    );
    fireEvent.click(getByLabelText(filters[0].options[0].label));
    expect(onChange.mock.calls).toMatchSnapshot();
  });

  it("renders correct summaryLabel if filter.label is an empty string", () => {
    const filters = [
      {
        label: "",
        name: "product-1",
        value: ["option-2"],
        options: [
          {
            label: "Option 1",
            value: "option-1"
          },
          {
            label: "Option 2",
            value: "option-2",
            isDisabled: true
          }
        ]
      }
    ];

    const { container } = render(<Filters filters={filters} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
