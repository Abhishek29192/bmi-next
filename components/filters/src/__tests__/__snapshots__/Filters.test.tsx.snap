// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component calls onChange handler 1`] = `
Array [
  Array [
    "product-1",
    "option-1",
    true,
  ],
]
`;

exports[`Filters component renders correctly 1`] = `
<div
  class="Filters"
>
  <div
    class="Accordion Accordion--no-inner-padding"
  />
</div>
`;

exports[`Filters component renders disabled filters 1`] = `
<div
  class="Filters"
>
  <div
    class="Accordion Accordion--no-inner-padding"
  >
    <div
      class="MuiPaper-root MuiAccordion-root item MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        aria-label="Label 1"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Label 1
          </h6>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root details"
              >
                <div
                  class="list"
                >
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-12"
                              data-indeterminate="false"
                              name="option-1"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 1
                        </span>
                      </label>
                    </div>
                  </div>
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root Mui-disabled"
                      >
                        <span
                          aria-disabled="true"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-10 Mui-checked PrivateSwitchBase-disabled-11 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
                          tabindex="-1"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              checked=""
                              class="PrivateSwitchBase-input-12"
                              data-indeterminate="false"
                              disabled=""
                              name="option-2"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
                        >
                          Option 2
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters component renders multiple filters 1`] = `
<div
  class="Filters"
>
  <div
    class="Accordion Accordion--no-inner-padding"
  >
    <div
      class="MuiPaper-root MuiAccordion-root item MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        aria-label="Label 1"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Label 1
          </h6>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root details"
              >
                <div
                  class="list"
                >
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-4"
                              data-indeterminate="false"
                              name="option-1"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 1
                        </span>
                      </label>
                    </div>
                  </div>
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-4"
                              data-indeterminate="false"
                              name="option-2"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 2
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root item MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        aria-label="Label 2"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Label 2
          </h6>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root details"
              >
                <div
                  class="list"
                >
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-4"
                              data-indeterminate="false"
                              name="option-1"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 1
                        </span>
                      </label>
                    </div>
                  </div>
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-4"
                              data-indeterminate="false"
                              name="option-2"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 2
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters component renders selected filters 1`] = `
<div
  class="Filters"
>
  <div
    class="Accordion Accordion--no-inner-padding"
  >
    <div
      class="MuiPaper-root MuiAccordion-root item MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        aria-label="Label 1"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Label 1
          </h6>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root details"
              >
                <div
                  class="list"
                >
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-8"
                              data-indeterminate="false"
                              name="option-1"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 1
                        </span>
                      </label>
                    </div>
                  </div>
                  <div>
                    <div
                      class="MuiFormControl-root Checkbox"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              checked=""
                              class="PrivateSwitchBase-input-8"
                              data-indeterminate="false"
                              name="option-2"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Option 2
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
